libminifi/src/processors/FocusArchiveEntry.cpp

```diff
@@ -0,0 +1,340 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "processors/FocusArchiveEntry.h"
+
+#include <archive.h>
+#include <archive_entry.h>
+
+#include <string.h>
+
+#include <boost/filesystem.hpp>
+
+#include <string>
+#include <set>
+
+#include <iostream>
+#include <fstream>
+#include <memory>
+
+#include "core/ProcessContext.h"
+#include "core/ProcessSession.h"
+
+#include "json/json.h"
+#include "json/writer.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+core::Property FocusArchiveEntry::Path(
+    "Path",
+    "The path within the archive to focus (\"/\" to focus the total archive)",
+    "");
+core::Relationship FocusArchiveEntry::Success(
+    "success",
+    "success operational on the flow record");
+
+bool FocusArchiveEntry::set_del_or_update_attr(std::shared_ptr<core::FlowFile> flowFile, const std::string key, std::string* value) const {
+  if (value == nullptr)
+    return flowFile->removeAttribute(key);
+  else if (flowFile->updateAttribute(key, *value))
+    return true;
+  else
+    return flowFile->addAttribute(key, *value);
+}
+
+void FocusArchiveEntry::initialize() {
+  //! Set the supported properties
+  std::set<core::Property> properties;
+  properties.insert(Path);
+  setSupportedProperties(properties);
+  //! Set the supported relationships
+  std::set<core::Relationship> relationships;
+  relationships.insert(Success);
+  setSupportedRelationships(relationships);
+}
+
+void FocusArchiveEntry::onTrigger(core::ProcessContext *context,
+                                  core::ProcessSession *session) {
+  auto flowFile = session->get();
+  std::shared_ptr<FlowFileRecord> flowFileRecord = std::static_pointer_cast<FlowFileRecord>(flowFile);
+
+  if (!flowFile) {
+    return;
+  }
+
+  std::string targetEntry;
+  context->getProperty(Path.getName(), targetEntry);
+
+  // Extract archive contents
+  ArchiveMetadata archiveMetadata;
+  archiveMetadata.focusedEntry = targetEntry;
+  ReadCallback cb(&archiveMetadata);
+  session->read(flowFile, &cb);
+
+  // For each extracted entry, import & stash to key
+  std::string targetEntryStashKey;
+
+  for (auto &entryMetadata : archiveMetadata.entryMetadata) {
+    if (entryMetadata.entryType == AE_IFREG) {
+      logger_->log_info("FocusArchiveEntry importing %s from %s",
+          entryMetadata.entryName.c_str(),
+          entryMetadata.tmpFileName.c_str());
+      session->import(entryMetadata.tmpFileName, flowFile, false, 0);
+      char stashKey[37];
+      uuid_t stashKeyUuid;
```

Comment:
Is there any reason you were unable to use the ID generator?

Reply:
Fixed in d2e7e34ab8b331ac484b9b16bd51455799a1502b